bindings/pydeck/pydeck/bindings/warnings.py

```diff
@@ -0,0 +1,10 @@
+import warnings
+
+
+def warn_for_mapbox_key():
+    warnings.warn(
+        ("A Mapbox API key is not set, which will lead to a blank base map. "
```

Comment:
1) I need the widget to be enabled to have back-channel communication but the library is usable without it, so I think I'd need to do both warning in the Python side and the generalized JS (@deck.gl/json) side. I'll include the JS in a separate PR.
2) Right, I need to include this.

Reply:
> but the library is usable without it,

Having to duplicate stuff is annoying. One way to avoid this: you could implement a mode where the JS side shows such errors in a closable div if there is no back channel.

Reply:
I think @ajduberstein is referring to the `to_html` code? When the code is run in an environment that doesn't support widgets, such as Colab, the JS is run in an iframe... is that connected to the widget at all? Without exception checking on the Python side, is there the possibility of a backchannel?

Reply:
What I am suggesting is that if there is no back-channel, show errors/warnings in an error message "div" instead of sending them over the back-channel.

The error popup could close after a timer expires or when a close button (x) is pressed.